<launch>
  <arg name="dataset" default="uhumans1" />
  <arg name="launch_voxblox" default="true" />

  <include file="$(find kimera_pgmo)/launch/generate_mesh.launch" pass_all_args="true"/>

  <node type="kimera_pgmo_node" name="kimera_pgmo_node" pkg="kimera_pgmo" output="screen" >
    <rosparam file="$(find kimera_pgmo)/params/$(arg dataset)/parameters.yaml" />

    <param name="output_prefix" value="/home/yunchang/Downloads/optimized" />

    <remap from="~input_mesh" to="/converted_mesh" />
    <remap from="~pose_graph_incremental" to="/kimera_vio_ros/pose_graph_incremental" />
  </node>

  <include file="$(find pose_graph_tools)/launch/posegraph_view.launch" >
    <arg name="frame_id"      value="world" />
    <arg name="graph_topic"   value="pose_graph" />
    <arg name="ns"            value="kimera_pgmo_node"/>
  </include>

</launch>